graveler/value.go

```diff
@@ -0,0 +1,88 @@
+package graveler
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+)
+
+/*
+ * Value is serialized in a trivial fixed-order format:
+ *
+ *    | len(Identity) | Identity | len(Value) | Value |
+ *
+ * where each length is serialized as a varint and additional bytes after Value are silently
+ * ignored.
+ */
+
+func writeVarint(w io.Writer, i int) error {
+	e := make([]byte, binary.MaxVarintLen64)
+	l := binary.PutVarint(e, int64(i))
+	_, err := w.Write(e[:l])
+	return err
+}
+
+func writeBytes(tag string, w io.Writer, b []byte) error {
+	if err := writeVarint(w, len(b)); err != nil {
+		return fmt.Errorf("write %s length: %w", tag, err)
+	}
+	if _, err := w.Write(b); err != nil {
+		return fmt.Errorf("write %s: %w", tag, err)
+	}
+	return nil
+}
+
+// MarshalValue returns bytes that uniquely unmarshal into a Value equal to v.
+func MarshalValue(v *Value) ([]byte, error) {
+	var w = &bytes.Buffer{}
+
+	if err := writeBytes("identity", w, v.Identity); err != nil {
```

Comment:
Assuming we have exactly 2 tags that are constant, could be better to define them as such?

Reply:
Not sure I follow.  But the only purpose of the `tag` argument is to report errors -- obviously we can do it here outside, so removing that.